### PR TITLE
Gw fcoder

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/GWF/GWFcoder/default.schema.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "title": "Actuality duration",
+      "type": "integer",
+      "unit": "min",
+      "description": "Actuality duration in min",
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "volume": {
+      "title": "Volume",
+      "unit": "m3",
```
Comment (mariusgiger) [inline types/GWF/GWFcoder/default.schema.json:23]: write m^3?

File: types/GWF/GWFcoder/lifecycle.schema.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      "title": "commandError2",
+      "description": "Command error 2",
+      "type": "boolean"
+    },
+    "commandError3": {
+      "title": "commandError3",
+      "description": "Command error 3",
+      "type": "boolean"
+    },
+
```
Comment (mariusgiger) [inline types/GWF/GWFcoder/lifecycle.schema.json:69]: ;)

File: types/GWF/GWFcoder/lifecycle.schema.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      "description": "Backflow",
+      "type": "boolean"
+    },
+    "noUsage": {
+      "title": "No usage",
+      "description": "No usage",
+      "type": "boolean"
+    },
+    "loraLinkError": {
+      "title": "Lora link error",
```
Comment (mariusgiger) [inline types/GWF/GWFcoder/lifecycle.schema.json:96]: write LoRa

File: types/GWF/GWFcoder/lifecycle.schema.json [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      "description": "Lora link error",
+      "type": "boolean"
+    },
+    "batteryLifetime": {
+      "title": "Battery lifetime",
+      "type": "integer",
+      "unit": "semester",
+      "description": "Battery lifetime in semester",
```
Comment (mariusgiger) [inline types/GWF/GWFcoder/lifecycle.schema.json:104]: better use month?

**Review by mariusgiger — COMMENTED** [review]
looks clean, some minor comments

File: types/GWF/GWFcoder/lifecycle.schema.json (hunk 4 above)
Comment (J2thatsme) [reply types/GWF/GWFcoder/lifecycle.schema.json:104]: so 4 Months per semester ?

Comment (mariusgiger) [reply types/GWF/GWFcoder/lifecycle.schema.json:103]: I think a semester is 6 months, what is written in the docs?

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**